src/Plugin/DataType/LoggerEntry.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\DataType\LoggerEntry.
+ */
+
+namespace Drupal\rules\Plugin\DataType;
+
+use Drupal\Core\TypedData\DataDefinitionInterface;
+use Drupal\Core\TypedData\Plugin\DataType\Map;
+use Drupal\Core\TypedData\TypedData;
+use Drupal\Core\TypedData\TypedDataInterface;
+
+/**
+ * The logger entry data type.
+ *
+ * @see: https://www.drupal.org/node/2625238
+ *
+ * @ingroup typed_data
+ *
+ * @DataType(
+ *   id = "loggerentry",
+ *   label = @Translation("Logger Entry"),
+ *   definition_class = "\Drupal\Rules\TypedData\LoggerEntryDataDefinition"
```

Comment:
This should be our own class that extends MapDataDefinition.php and implements getPropertyDefinitions().